docs/topics/asset/ERC1155721.md

```diff
@@ -0,0 +1,63 @@
+---
+breaks: false
+
+description: Custom dual ERC1155 and ERC721 implementation
+---
+
+# [Dual ERC1155 and ERC721 Implementation](https://github.com/thesandboxgame/sandbox-smart-contracts/blob/master/src/solc_0.8/asset/ERC1155ERC721.sol)
+
+## Introduction
+
+This document describes our custom implementation of both [EIP-1155](https://eips.ethereum.org/EIPS/eip-1155) (for limited editions tokens) and [EIP-721](https://eips.ethereum.org/EIPS/eip-721) (for non fungible, unique tokens) in one smart contract.
+
+One advantage of this dual implementation is that our token is compatible with wallets or marketplaces that support both EIP.
+
+Reading the [EIP-1155](https://eips.ethereum.org/EIPS/eip-1155) and [EIP-721](https://eips.ethereum.org/EIPS/eip-721) documentation is a pre-requisite to read this document, but
+here below we explain the most important aspects.
+
+### ERC721
+
+1. Non-fungible tokens (NFTs) are digital assets that are provably unique, creating digital scarcity.
+2. They can't be duplicated or divided.
+3. They have many use cases, including for digital collectibles, music, artwork, and in-game tokens.
+
+### ERC115
+
+1. Tokens standards like ERC-20 and ERC-721 require a separate contract to be deployed for each token type or collection, **ERC1155** allow you to use one contract for every ERC20 and ERC721 contract you may have.
+2. New functionality is possible with this design such as transferring multiple token types at once.
+
+## Implementation
+
+The ERC1155ERC721 contract shares the same functionalities as ERC20 and ERC721 (ie get balance, get owner, burn, mint, batch burn, batch mint, approval...).
+
+Each token ID contains all the data of the current token (Creator, isNFT, number of token when minted, pack ID...). When a specific information of this Id is needed a mask is applied to fetch the wanted data.
+
+The contract provide you the ability to change an ERC1155 token to an ERC721 unique token using the `extractERC721From` function. For example, it is used when you want to associate [a catalyst and gems](../catalyst/catalyst.md) to an asset or if you want to sell an asset on Sandbox marketplace.
+
+## Model
+
+```plantuml
+title class diagram
+
+class ERC1155ERC721 {
```

Comment:
Can you follow this guideline for the diagrams https://plantuml.com/en/class-diagram#3644720244dd6c6a

Reply:
What do you mean?